ui/apps/platform/src/Containers/Clusters/DelegateScanning/Components/DelegatedRegistriesList.tsx

```diff
@@ -1,27 +1,71 @@
+// TODO: remove lint override after @typescript-eslint deps can be resolved to ^5.2.x
+/* eslint-disable react/prop-types */
 import React from 'react';
 import {
     Bullseye,
     Button,
     Card,
-    CardBody,
+    CardFooter,
     EmptyState,
     EmptyStateBody,
     Title,
 } from '@patternfly/react-core';
+import PlusCircleIcon from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
 
-import { DelegatedRegistry } from 'services/DelegatedRegistryConfigService';
+import {
+    DelegatedRegistry,
+    DelegatedRegistryCluster,
+} from 'services/DelegatedRegistryConfigService';
+import DelegatedRegistriesTable from './DelegatedRegistriesTable';
 
 type DelegatedRegistriesListProps = {
     registries: DelegatedRegistry[];
+    clusters: DelegatedRegistryCluster[];
+    selectedClusterId: string;
+    addRegistryRow: () => void;
+    deleteRow: (number) => void;
+    handlePathChange: (number, string) => void;
+    handleClusterChange: (number, string) => void;
+    // TODO: re-enable next type after @typescript-eslint deps can be resolved to ^5.2.x
+    // updateRegistriesOrder: (DelegatedRegistry[]) => void;
```

Comment:
For our learning, what is the error now? And why will upgrade remove it?

Reply:
The error is that typescript-eslint does not understand how to parse the "array of custom type" for the argument in this function sig.

I thought I saw an exact duplicate of this issue, but I have several similar errors in my browser history from when I was researching it, and the most common answer is "upgrade to TS 5"

Without it commented out, here is the error:
<img width="618" alt="Screen Shot 2023-08-17 at 3 07 21 PM" src="https://github.com/stackrox/stackrox/assets/715729/c2f5a11e-be12-49ca-8819-9adcbef068c8">